src/main/java/com/zaxxer/hikari/pool/HikariPool.java

```diff
@@ -442,7 +442,16 @@ private PoolEntry createPoolEntry()
 
          final long maxLifetime = config.getMaxLifetime();
          if (maxLifetime > 0) {
-            final long variance = maxLifetime > 60_000 ? ThreadLocalRandom.current().nextLong(10_000) : 0;
+        	 long inHours = TimeUnit.MILLISECONDS.toHours(maxLifetime);
+        	 
+        	 // Every hour will add a 10 second random factor for the maxlifetime
+        	 long randomIntervalInMillis = inHours * 10_000;
+        	 
+        	 // Support the up to 10 seconds spread out when the lifetime is less than 1 hour
+        	 if (inHours == 0) {
```

Comment:
That's still not quite right.  If ``maxLifetime`` is 30 seconds, then ``maxLifetime / 1000`` is 30ms.  That's not enough variance.

Conversely, if the ``maxLifetime`` is 30 *minutes*, and the ``maxPoolSize`` is 10, the variance is only 10 seconds.

Here is my proposal:
```java
long randomIntervalInMillis = Math.max(10_000, maxLifetime / 20);
```
Division by 20 is equal to 5% of the ``maxLifetime``.  This says, we want connections to reach their max. lifetime within 5% variance of the specified value.  Some common cases:
```
  3 minutes = 15 seconds variance
  5 minutes = 30 seconds variance
 30 minutes = 90 seconds variance
      1 day = 71 minutes variance
```